File: app/src/main/java/com/autopreference/sample/MainActivity.java
package com.autopreference.sample;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;

import com.autopreference.sample.screens.DefaultSharedPreferencesFragment;
import com.autopreference.sample.screens.UserPreferenceFragment;

public final class MainActivity extends AppCompatActivity {

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setContentView(R.layout.activity_main);
	}

	public void onOpenUserSharedPreferencesButtonClicked(@NonNull View view) {
		SingleFragmentActivity.start(this, UserPreferenceFragment.TAG);
	}

	public void onOpenDefaultSharedPreferencesButtonClicked(@NonNull View view) {
		SingleFragmentActivity.start(this, DefaultSharedPreferencesFragment.TAG);
	}
}

File: app/src/main/java/com/autopreference/sample/MainActivity.kt
package com.autopreference.sample

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.autopreference.sample.screens.DefaultSharedPreferencesFragment
import com.autopreference.sample.screens.UserPreferenceFragment

internal class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
    }

    fun onOpenUserSharedPreferencesButtonClicked(view: View) {
        SingleFragmentActivity.start(this, UserPreferenceFragment.TAG)
    }

    fun onOpenDefaultSharedPreferencesButtonClicked(view: View) {
        SingleFragmentActivity.start(this, DefaultSharedPreferencesFragment.TAG)
    }
}
